src/EnergyPlus/PluginManager.cc

```diff
@@ -1459,6 +1469,22 @@ void PluginManager::runSingleUserDefinedPlugin([[maybe_unused]] EnergyPlusData &
 }
 #endif
 
+int PluginManager::getUserDefinedCallbackIndex(EnergyPlusData &state, const std::string &callbackProgramName)
+{
+    for (int i = 0; i < state.dataPluginManager->userDefinedCallbackNames.size(); i++) {
+        if (state.dataPluginManager->userDefinedCallbackNames[i] == callbackProgramName) {
+            return i;
+        }
+    }
+    return -1;
+}
```

Comment:
User defined components can get the working index for their callback by name using this worker function.